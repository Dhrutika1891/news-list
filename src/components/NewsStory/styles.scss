.news-story {
  display: grid;
  grid-template-columns: 50px auto;
  gap: 20px;
  padding: 10px;
  border: 1px solid grey;
  width: 100%;

  &__image {
    height: 50px;
    width: 50px;
  }

  &__container {
    display: grid;
    gap: 5px;
    > h3,
    h6 {
      margin-bottom: 0px;
    }
  }
}
