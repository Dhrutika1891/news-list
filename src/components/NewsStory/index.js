import React, { useState } from "react";

import "./styles.scss";

const NewsStory = ({ story }) => {
  const baseClass = "news-story";
  const [isError, setIsError] = useState(false);
  return (
    <div className={baseClass} data-testid="news-story">
      <img
        className={`${baseClass}__image`}
        data-testid="news-story-image"
        src={isError ? "https://picsum.photos/200" : story?.imageURL}
        alt={"News"}
        onError={() => {
          setIsError(true);
        }}
      />
      <div
        className={`${baseClass}__container`}
        data-testid="news-story-container"
      >
        <h3 data-testid="news-story-title">{story?.title}</h3>
        <h6 data-testid="news-story-subtitle">{story?.subtitle}</h6>
        <div data-testid="news-story-detail">{story?.newsDetail}</div>
        <a
          data-testid="news-story-link"
          target="_blank"
          rel="noreferrer"
          href={story?.newsURL}
        >
          Read more...
        </a>
      </div>
    </div>
  );
};

export default NewsStory;
