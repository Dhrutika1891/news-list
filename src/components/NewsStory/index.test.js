import { render } from "@testing-library/react";
import React from "react";

import NewsStory from ".";

describe("component news story", () => {
  it("should render news story", () => {
    expect.hasAssertions();

    const { getByTestId, baseElement } = render(<NewsStory status="loading" />);

    const newsStory = getByTestId("news-story");
    const newsImage = getByTestId("news-story-image");
    const newsContainer = getByTestId("news-story-container");
    const newsTitle = getByTestId("news-story-title");
    const newsSubtitle = getByTestId("news-story-subtitle");
    const newsDetail = getByTestId("news-story-detail");
    const newsLink = getByTestId("news-story-link");

    expect(newsStory).toBeInTheDocument();
    expect(newsImage).toBeInTheDocument();
    expect(newsContainer).toBeInTheDocument();
    expect(newsTitle).toBeInTheDocument();
    expect(newsSubtitle).toBeInTheDocument();
    expect(newsDetail).toBeInTheDocument();
    expect(newsLink).toBeInTheDocument();

    expect(baseElement).toMatchSnapshot();
  });
});
