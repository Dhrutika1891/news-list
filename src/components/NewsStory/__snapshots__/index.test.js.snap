// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component news story should render news story 1`] = `
<body>
  <div>
    <div
      class="news-story"
      data-testid="news-story"
    >
      <img
        alt="News"
        class="news-story__image"
        data-testid="news-story-image"
      />
      <div
        class="news-story__container"
        data-testid="news-story-container"
      >
        <h3
          data-testid="news-story-title"
        />
        <h6
          data-testid="news-story-subtitle"
        />
        <div
          data-testid="news-story-detail"
        />
        <a
          data-testid="news-story-link"
          rel="noreferrer"
          target="_blank"
        >
          Read more...
        </a>
      </div>
    </div>
  </div>
</body>
`;
